File: python code/4_Variable_transformation.py
#----------------------------------------------------------------------------------
# Varaible Transformation
#----------------------------------------------------------------------------------

# crime rate
sns.jointplot(x='crime_rate',y='price',data=df) #before follows log graph concept
df.crime_rate=np.log(1+df.crime_rate)
sns.jointplot(x='crime_rate',y='price',data=df) #after follows log graph concept

# dist 1,2,3,4 collab as avg
df['avg_dist']=(df.dist1+df.dist2+df.dist3+df.dist4)/4
del df['dist1']
del df['dist2']
del df['dist3']
del df['dist4']

new_describe_table=df.describe()

File: R code/4_Variable_transformation.r
# Variable transformation

#crime rate
plot(df$price,df$crime_rate)#before VT
df$crime_rate=log(1+df$crime_rate)
plot(df$price,df$crime_rate)#after VT

# avg distance
df$avg_dist=(df$dist1+df$dist2+df$dist3+df$dist4)/4

df2=df[,-7:-10]

df<-df2
rm(df2)
